app/models/user.rb

```diff
@@ -1,4 +1,7 @@
 class User < ActiveRecord::Base
+    validates :email, :provider, :uid, :token,  presence: true
+    validates_format_of :email, :with => /@/
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer the new style validations `validates :column, format: value` over `validates_format_of`.